### PR TITLE
Add profiler TLAB support.

### PR DESCRIPTION
**Pull request by breedx-splk**
Tracking allocations and the stacks for where they happen.

### CODE REVIEW

File: profiler/README.md [hunk 1]
```diff
@@ -13,3 +13,4 @@ It should be considered experimental and is completely unsupported.
 |`splunk.profiler.keep-files`         | false                  | leave JFR files on disk id `true`         |
 |`splunk.profiler.logs-endpoint`      | http://localhost:4317  | where to send OTLP logs                   |
 |`splunk.profiler.period.{eventName}` | n/a                    | customize period (in ms) for a specific jfr event. For example, to set the ThreadDump frequency to 1s (100ms): `-Dsplunk.profiler.period.threaddump=1000` |
+|`splunk.profiler.tlab.enabled`       | true                   | set to `false` to disable TLAB memory events |
```
Comment (jkwatson) [inline profiler/README.md:16]: don't we want this off by default, due to the cost?

File: profiler/src/main/java/com/splunk/opentelemetry/profiler/util/StackSerializer.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  }
+
+  public String serialize(RecordedStackTrace stack) {
+    List<RecordedFrame> frames = stack.getFrames();
+    int skipNum = Math.max(0, frames.size() - maxDepth);
+    return frames.stream()
+        .skip(skipNum)
+        .limit(maxDepth)
+        .reduce(
+            "",
```
Comment (mateuszrzeszutek) [inline profiler/src/main/java/com/splunk/opentelemetry/profiler/util/StackSerializer.java:44]: Nit: would it be possible to use a StringBuilder here?

**Review by mateuszrzeszutek — APPROVED** [review]

File: profiler/README.md (hunk 1 above)
Comment (breedx-splk) [reply profiler/README.md:16]: I think this is still an open question. We haven't yet really quantified the cost yet, so maybe to play it safe is a better idea. What do @iNikem and @mateuszrzeszutek and @laurit think?

File: profiler/src/main/java/com/splunk/opentelemetry/profiler/util/StackSerializer.java (hunk 2 above)
Comment (breedx-splk) [reply profiler/src/main/java/com/splunk/opentelemetry/profiler/util/StackSerializer.java:44]: Good idea! Thanks.

File: profiler/README.md (hunk 1 above)
Comment (jkwatson) [reply profiler/README.md:16]: My vote would be to leave it off by default until we've at least verified it's not going to destroy the average app's performance.

Comment (breedx-splk) [reply profiler/README.md:16]: It's still turned off unless `splunk.profiler.enabled=true`, which is not the default (yet?).

Comment (breedx-splk) [reply profiler/README.md:16]: Ok, I think you're right and that this is a better idea, so I've inverted the logic around this and defaulted tlab support to off.

File: profiler/src/main/java/com/splunk/opentelemetry/profiler/util/StackSerializer.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    List<RecordedFrame> frames = stack.getFrames();
+    int skipNum = Math.max(0, frames.size() - maxDepth);
+    return frames.stream()
+        .skip(skipNum)
+        .limit(maxDepth)
+        .reduce(
+            new StringBuilder(),
+            (sb, recordedFrame) -> {
+              String frameStr = serializeFrame(recordedFrame);
+              return sb.length() == 0 ? sb.append(frameStr) : sb.append("\n").append(frameStr);
```
Comment (mateuszrzeszutek) [inline profiler/src/main/java/com/splunk/opentelemetry/profiler/util/StackSerializer.java:47]: You can use the `StringBuilder` in `serializeFrame` too 😄 

```java
if (sb.length() > 0) {
  sb.append("\n");
}
serializeFrame(recordedFrame, sb);
return sb;
```

And avoid one string concatenation per each frame.

**Review by mateuszrzeszutek — APPROVED** [review]

File: profiler/src/main/java/com/splunk/opentelemetry/profiler/util/StackSerializer.java (hunk 3 above)
Comment (breedx-splk) [reply profiler/src/main/java/com/splunk/opentelemetry/profiler/util/StackSerializer.java]: Sounds good.

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**